From: Anders Larsen <al@alarsen.net>
Date: Tue, 26 Jan 2010 18:04:34 +0100
Subject: [PATCH] make wxSocketBase::Peek() behave intuitively
 temporarily set wxSOCKET_NOWAIT so Peek() won't block if no data is available

Signed-off-by: Anders Larsen <al@alarsen.net>
---
 src/common/socket.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/common/socket.cpp b/src/common/socket.cpp
index 44b408c..ee7057a 100644
--- a/src/common/socket.cpp
+++ b/src/common/socket.cpp
@@ -1109,6 +1109,8 @@ wxSocketBase& wxSocketBase::Peek(void* buffer, wxUint32 nbytes)
 {
     wxSocketReadGuard read(this);
 
+    wxSocketWaitModeChanger changeFlags(this, wxSOCKET_NOWAIT);
+
     m_lcount = DoRead(buffer, nbytes);
 
     Pushback(buffer, m_lcount);
-- 
1.5.4.3

